api/server.go

```diff
@@ -20,7 +20,7 @@ import (
 var indexerDb idb.IndexerDb
 
 // Serve starts an http server for the indexer API. This call blocks.
-func Serve(ctx context.Context, serveAddr string, db idb.IndexerDb, log *log.Logger, tokens []string, developerMode bool) {
+func Serve(ctx context.Context, serveAddr string, db idb.IndexerDb, fetcherError error, log *log.Logger, tokens []string, developerMode bool) {
```

Comment:
was confused for a minute because I expected a function pointer or interface that _returned_ `error` so that it could return `nil` if there was no error yet. Then I got to Fetcher _implementing_ error and I get it now. My bias says `err == nil` is better than `foo.Error() == ""`, but probably this is fine.

Reply:
Yeah, this is a total hack... probably some sort of error registry would be better